tests/instrumentation-framework/src/main/java/org/hawkular/apm/tests/dockerized/TestScenarioRunner.java

```diff
@@ -0,0 +1,230 @@
+     * @param apmServerPort - port at which Hawkular APM server will be started
+     * @throws DockerCertificateException
+     */
+    public TestScenarioRunner(String apmVersion, int apmServerPort) throws DockerCertificateException {
+        this.apmVersion = apmVersion;
+        this.apmServerPort = apmServerPort;
+
+        this.objectMapper = new ObjectMapper();
+        apmServer = new ApmMockServer();
+        apmServer.setPort(apmServerPort);
+        apmServer.setShutdownTimer(60*60*1000);
+    }
+
+    /**
+     * Run a test scenario
+     *
+     * @param testScenario
+     * @return number of successful test cases
+     */
+    public int run(TestScenario testScenario) {
+        if (testScenario.getEnvironment().getDockerCompose() != null &&
+                testScenario.getEnvironment().getImage() != null) {
+            throw new IllegalArgumentException("Ambiguous environment: defined docker image and docker-compose");
+        }
+
+        System.out.println("Starting test scenario: " + testScenario);
+        int successfulTestCases = 0;
+
+        for (TestCase test: testScenario.getTests()) {
+            if (test.isSkip()) {
+                continue;
+            }
+
+            TestEnvironmentExecutor testEnvironmentExecutor = testEnvironmentExecutor(testScenario);
+
+            try {
+                runTestCase(testScenario, test, testEnvironmentExecutor);
+                successfulTestCases++;
+            } catch (TestFailException ex) {
+                System.out.println("Test case failed: " + ex.toString());
+                System.out.println(ex.getMessage());
+                ex.printStackTrace();
+            }
+
+            testEnvironmentExecutor.close();
+        }
+
+        return successfulTestCases;
+    }
+
+    /**
+     * Run single test case
+     *
+     * @param testScenario Test scenario of the test case
+     * @param testCase Test case to run
+     * @param testEnvironmentExecutor Initialized environment
+     * @throws TestFailException
+     */
+    private void runTestCase(TestScenario testScenario, TestCase testCase,
+                             TestEnvironmentExecutor testEnvironmentExecutor) throws TestFailException {
+
+        System.out.println("Executing test case: " + testCase);
+
+        String environmentId = null;
+        try {
+            if (testEnvironmentExecutor instanceof DockerComposeExecutor) {
+                ((DockerComposeExecutor) testEnvironmentExecutor).createNetwork();
+            }
+
+            /**
+             * Start APM server
+             */
+            apmServer.setHost("0.0.0.0");
+            apmServer.setTraces(new ArrayList<>());
+            apmServer.run();
+
+            /**
+             * Run a container and wait
+             */
+            environmentId = testEnvironmentExecutor.run(testScenario.getEnvironment());
+            Thread.sleep(testScenario.getEnvironment().getInitWaitSeconds()*1000);
```

Comment:
The environment in this case always docker container there is some app which has to start. `waitInitSeconds` defines wait time for that application to start and fully initialize. Therefore, this time is configurable in test definition file. Method `run` exists no matter the application initialized.

Life cycle of the test case:
1. start environment (which includes start of the app)
2. wait for app start
3. run action in the app which is executed inside docker container.
4. wait some time that the action is processed and some data propagated to apm server
5. verify results.

Is it ok?